### PR TITLE
[core] Introduce TableCommit.abort

### PR DESCRIPTION
**Pull request by JingsongLi**
### Purpose

Introduce TableCommit.abort, somewhere needs this method to abort job commit, like Hive write.

### Tests

`FileStoreTableTestBase.testAbort`

### API and Format 

Add API

### Documentation

Not in normal path, don't need to document this.

### CODE REVIEW

**Review by yuzelin — APPROVED** [review]
LGTM

File: paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommit.java [hunk 1]
```diff
@@ -72,4 +73,7 @@ void overwrite(
      * @param partitions A list of partition {@link Map}s. NOTE: cannot be empty!
      */
     void dropPartitions(List<Map<String, String>> partitions, long commitIdentifier);
+
+    /** Abort an unsuccessful commit. The data files will be deleted. */
+    void abort(List<CommitMessage> commitMessages);
```
Comment (SteNicholas) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommit.java:78]: Could the param use `ManifestCommittable committable`?

Comment (JingsongLi) [reply paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommit.java:78]: `ManifestCommittable` need an identifier. But here we don't need that.

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Pull request merged**